src/table/components/head/HeadCellMenu.tsx

```diff
@@ -46,15 +65,89 @@ export default function HeadCellMenu({ column, tabIndex }: HeadCellMenuProps) {
+                },
+                icon: <SelectAll />,
+                enabled: selectionAnalysis.canSelectAll,
+              },
+              {
+                id: 2,
+                itemTitle: translator.get('SNTable.MenuItem.ClearSelections'),
+                onClick: async () => {
+                  await fieldInstance?.clear();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <ClearSelections />,
+                enabled: selectionAnalysis.clearSelections,
+              },
+              {
+                id: 3,
+                itemTitle: translator.get('SNTable.MenuItem.SelectPossible'),
+                onClick: async () => {
+                  await fieldInstance?.selectPossible();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectPossible />,
+                enabled: selectionAnalysis.canSelectPossible,
+              },
+              {
+                id: 4,
+                itemTitle: translator.get('SNTable.MenuItem.SelectAlternative'),
+                onClick: async () => {
+                  await fieldInstance?.selectAlternative();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectAlternative />,
+                enabled: selectionAnalysis.canSelectAlternative,
+              },
+              {
+                id: 5,
+                itemTitle: translator.get('SNTable.MenuItem.SelectExcluded'),
+                onClick: async () => {
+                  await fieldInstance?.selectExcluded();
+                  setOpenMenuDropdown(false);
+                },
+                icon: <SelectExcluded />,
+                enabled: selectionAnalysis.canSelectExcluded,
               },
             ],
           ]
         : []),
     ],
-    [translator, showSearchMenuItem]
+    [translator, showSearchMenuItem, fieldInstance, selectionAnalysis]
   );
 
+  const checkStateCountByKey = <T,>(keys: (keyof T)[], obj: T): boolean => {
+    return keys.some((key) => obj[key] > 0);
+  };
+
+  const runSelectionAnalysis = (layout: TableLayout) => {
+    const dimInfo = layout.qHyperCube.qDimensionInfo.find((dim) => dim.qFallbackTitle === column.label);
+    if (!dimInfo) return;
+    setSelectionAnalysis({
+      canSelectAll: checkStateCountByKey(['qOption', 'qAlternative', 'qExcluded', 'qDeselected'], dimInfo.qStateCounts),
+      clearSelections: checkStateCountByKey(['qSelected'], dimInfo.qStateCounts),
+      canSelectPossible: checkStateCountByKey(['qOption'], dimInfo.qStateCounts),
+      canSelectAlternative: checkStateCountByKey(['qAlternative'], dimInfo.qStateCounts),
+      canSelectExcluded: checkStateCountByKey(['qAlternative', 'qExcluded'], dimInfo.qStateCounts),
+    });
+  };
+
+  const handleOpenDropdown = async () => {
+    if (!model) return;
+    setOpenMenuDropdown(!openMenuDropdown);
+    model.getLayout().then((l) => runSelectionAnalysis(l as TableLayout));
```

Comment:
So on first render, are you able to see that the options are all disabled? then some become enabled? If so, what happens when the next time you open it? I thinking there could be a problem with some options being enabled from last time it was open, and there is a minimal change to click something that has not been disabled yet

Reply:
And actually, we might only want to get the layout when opening, not when closing. Then maybe something like this
```
if (!openMenuDropdown) {
  const layout =  await model.getLayout();
  runSelectionAnalysis(l as TableLayout);
}
setOpenMenuDropdown(!openMenuDropdown);

Reply:
what happens now is, i really cant see any items disabled at the first render, because `model.getLayout()` call happens really fast. I could reproduce your idea when i had explicitly called `model.getLayout()` after 2 sec inside of a setTimeout. and in that case, i was initially seeing it disabled and after 2 sec and getLayout call i was seeing the items being enabled. 

So, I'm thinking may be, we need to disable them all when we close the popover (and menu items will be fine after trying to open popover again because of getLayout call)? what do you think?

Reply:
Yes looks like something like that could be needed. I guess you could as keep a "isLoading" state and always disable them when that is true.

Reply:
after implementing "disabling all when dropdown closes" it solved the issue. Now, every time you open dropdown -> call getLayout -> mean while items disabled (which is **not** noticeable) -> you get layout -> every thing is in order **and** when you close -> disabled them first -> then close dropdown => which literally fading dropdown and disabling happens at the same time and it's barely noticeable.

I think this cycle 👆 covers the need for `isLoading` state and that state would be needed in a scenario like we have dropdown open, and then selection changes -> which would require us to call getLayout **while** dropdown is still open.

So I think we don't need it until designs require us somehow! **OR** maybe we need it for the sub-menu case? (probably not either)